api/admin.py

```diff
@@ -175,3 +175,49 @@ def get_messages(message_filter: str = 'all'):
         code = OK
 
     return response, code
+
+# Route to get a singular message with given id.
+@admin.route('/get-message/<message_id>', methods=['GET'])
+def get_message(message_id: int):
+    """Get a singular message with the given id
+    }
+    """
+    response = dict()
+
+    # If the message exists, create a payload:
+    if message := ContactFormMessage.query.filter_by(id=message_id).first():
+        response['message'] = message.to_dict()
+        code = OK
+
+    else:
+        response['err_msg'] = f'Message with id: {message_id}\
+                                not found in database.'
+        code = BAD_REQUEST
+
+    return response, code
+
+
+
+#Route for seen/unseen messages
+@admin.route('seen_message/<message_id>', methods=['PUT'])
+def seen_message(message_id: int):
+    """Mark a message as seen
+
+    If a message has been seen, remove dot. 
+    """
+    response = dict()
+
+    # Check if listing is in database, then update and return to Jake:
+    if message := ContactFormMessage.query.filter_by(id=message_id).first():
+        message.was_seen = True if message.was_seen is False else False
```

Comment:
Looks like we have a copy paste error in the comment, and in the logic.  Unless I am wrong, do you want a seen message to be marked as unseen after hitting this route again?

Reply:
Good point. We don't want to just flip the boolean, we only want to change from True to False and never change again